selftests/unit/test_remote.py

```diff
@@ -25,154 +30,71 @@ class RemoteTestRunnerTest(unittest.TestCase):
+        """
+        Test RemoteTestRunner.run_suite()
+
+        The general idea of this test is to:
+
+        1) Create the machinery necessary to get a RemoteTestRunner
+           setup inside a job, or looking at it the other way around, to
+           have a runner that is created with a valid job.
+
+        2) Mock the interactions with a remote host.  This is done here
+           basically by mocking 'Remote' and 'fabric' usage.
+
+        3) Provide a polluted JSON to be parsed by the RemoteTestRunner
+
+        4) Assert that those results are properly parsed into the
+           job's result
+        """
+        job_args = argparse.Namespace(test_result_total=1,
+                                      remote_username='username',
+                                      remote_hostname='hostname',
+                                      remote_port=22,
+                                      remote_password='password',
+                                      remote_key_file=None,
+                                      remote_timeout=60,
+                                      show_job_log=False,
+                                      mux_yaml=['~/avocado/tests/foo.yaml',
+                                                '~/avocado/tests/bar/baz.yaml'],
+                                      dry_run=True,
+                                      env_keep=None,
+                                      reference=['/tests/sleeptest.py',
+                                                 '/tests/other/test',
+                                                 'passtest.py'])
+
+        job = Job(job_args)
+        runner = avocado_runner_remote.RemoteTestRunner(job, job.result)
+        runner.check_remote_avocado = mock.Mock(return_value=(True,
+                                                              (version.MAJOR,
+                                                               version.MINOR)))
+
+        # These are mocked at their source, and will prevent fabric from
+        # trying to contact remote hosts
+        with mock.patch('avocado_runner_remote.Remote'):
+            with mock.patch('avocado_runner_remote.fabric'):
+                runner.remote = avocado_runner_remote.Remote(job_args.remote_hostname)
+
+            # This is the result that the run_suite() will get from remote.run
+            remote_run_result = process.CmdResult()
+            remote_run_result.stdout = JSON_RESULTS
+            remote_run_result.exit_status = 0
+            runner.remote.run = mock.Mock(return_value=remote_run_result)
+
+            # We have to fake the uncompressing and removal of the zip
+            # archive that was never generated on the "remote" end
+            # This test could be expand by mocking creating an actual
+            # zip file instead, but it's really overkill
+            with mock.patch('avocado_runner_remote.archive.uncompress'):
+                with mock.patch('avocado_runner_remote.os.remove'):
```

Comment:
python2.7 and 3 supports `with foo as foo, bar as bar` and it'd probably look better here.

Reply:
I tried a syntax that would let me break the line without a `\`, and couldn't find it.  Without a line break, that is, something like this:

```python
diff --git a/selftests/unit/test_remote.py b/selftests/unit/test_remote.py
index 57719a9b..9ec3cddd 100644
--- a/selftests/unit/test_remote.py
+++ b/selftests/unit/test_remote.py
@@ -72,9 +72,8 @@ class RemoteTestRunnerTest(unittest.TestCase):
 
         # These are mocked at their source, and will prevent fabric from
         # trying to contact remote hosts
-        with mock.patch('avocado_runner_remote.Remote'):
-            with mock.patch('avocado_runner_remote.fabric'):
-                runner.remote = avocado_runner_remote.Remote(job_args.remote_hostname)
+        with mock.patch('avocado_runner_remote.Remote') as rem, mock.patch('avocado_runner_remote.fabric') as fab):
+            runner.remote = avocado_runner_remote.Remote(job_args.remote_hostname)
 
             # This is the result that the run_suite() will get from remote.run
             remote_run_result = process.CmdResult()
```

Would look worse IMO.